packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -279,12 +279,13 @@ export class ListLayout<T, O extends ListLayoutOptions = ListLayoutOptions> exte
       }
 
       // Build each loader that exists in the collection that is outside the visible rect so that they are persisted
-      // at the proper estimated location
-      if (y > this.requestedRect.maxY) {
+      // at the proper estimated location. If the node.type is "section" then we don't do this shortcut since we have to
+      // build the sections to see how tall they are.
+      if ((node.type === 'item' || node.type === 'loader') && y > this.requestedRect.maxY) {
```

Comment:
This is still basically how it worked previously, https://github.com/adobe/react-spectrum/blob/9786f69a1cb6b9a9158fcef4dbb9881faefac016/packages/%40react-stately/layout/src/ListLayout.ts#L275

We only used this shortcut when processing items, if there are sections then iterating through the collection will only return sections since we don't support sections and sectionless items at the same time